charts/ndb-operator/templates/deployment-ndb-operator-controller-manager.yaml

```diff
@@ -0,0 +1,106 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    control-plane: controller-manager
+  name: {{ include "ndb-operator.fullname" . }}-controller-manager
+  namespace: {{.Release.Namespace}}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      control-plane: {{ include "ndb-operator.fullname" . }}-controller-manager
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        control-plane: {{ include "ndb-operator.fullname" . }}-controller-manager
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end}}
+      serviceAccountName: {{ include "ndb-operator.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+
+      - args:
+        - --health-probe-bind-address=:8081
+        - --metrics-bind-address=127.0.0.1:8080
+        - --leader-elect
+        image: {{ .repository }}:{{ .tag | default $.Chart.AppVersion }}
```

Comment:
node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason         Age   From               Message
  ----     ------         ----  ----               -------
  Normal   Scheduled      13s   default-scheduler  Successfully assigned ndb-operator-system/ndb-operator-controller-manager-74c8776d8b-xhlvq to lima-rancher-desktop
  Warning  FailedMount    13s   kubelet            MountVolume.SetUp failed for volume "cert" : secret "webhook-server-cert" not found
  Warning  InspectFailed  12s   kubelet            Failed to apply default image tag ":v0.0.7": couldn't parse image reference ":v0.0.7": invalid reference format
  Warning  Failed         12s   kubelet            Error: InvalidImageName
  Normal   Pulling        12s   kubelet            Pulling image "gcr.io/kubebuilder/kube-rbac-proxy:v0.14.1"
ndb-operator $

Reply:
The image URL is derived by templating logic: image: {{ .repository }}:{{ .tag | default $.Chart.AppVersion }}

inside values.yaml :

```
image:
  repository: ghcr.io/nutanix-cloud-native/ndb-operator/controller
  pullPolicy: IfNotPresent
  # Overrides the image tag whose default is the chart appVersion.
  tag: ""
```

inside Chart.yaml
```
version: 0.0.7
appVersion: "v0.0.7"
```

Based on that the value for imageUrl is resolved to 
`ghcr.io/nutanix-cloud-native/ndb-operator/controller:v0.0.7`

But the helm install is failing with "`InvalidImageName`"

Reply:
This may be due to tag v0.0.7 is not available. https://github.com/nutanix-cloud-native/ndb-operator/tags

In this case, what should be the course of action for testing the helm install? Should we need to create a tag first and then test this out?

Reply:
Removing the colon ":" gives different error:

```
Events:
  Type     Reason       Age              From               Message
  ----     ------       ----             ----               -------
  Normal   Scheduled    17s              default-scheduler  Successfully assigned ndb-operator-system/ndb-operator-controller-manager-57bf7c8f9b-6vcks to lima-rancher-desktop
  Warning  FailedMount  18s              kubelet            MountVolume.SetUp failed for volume "cert" : secret "webhook-server-cert" not found
  Normal   Pulling      16s              kubelet            Pulling image "v0.0.7"
  Warning  Failed       5s               kubelet            Failed to pull image "v0.0.7": rpc error: code = Unknown desc = Error response from daemon: pull access denied for v0.0.7, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
  Warning  Failed       5s               kubelet            Error: ErrImagePull
  Normal   Pulled       5s               kubelet            Container image "gcr.io/kubebuilder/kube-rbac-proxy:v0.14.1" already present on machine
  Normal   Created      5s               kubelet            Created container kube-rbac-proxy
  Normal   Started      5s               kubelet            Started container kube-rbac-proxy
  Normal   BackOff      3s (x2 over 4s)  kubelet            Back-off pulling image "v0.0.7"
  Warning  Failed       3s (x2 over 4s)  kubelet            Error: ImagePullBackOff
```